### PR TITLE
[MRG] Adding brief descriptions and `-h/--help` text to sourmash `gather`, `search`, and `compare`

### PR DESCRIPTION
**Pull request by keyabarve**
Addresses #1557

What was done:
* Taking information from the command-line docs, https://sourmash.readthedocs.io/en/latest/command-line.html,
    * create a draft 'usage' string as in https://github.com/dib-lab/sourmash/blob/latest/src/sourmash/cli/compute.py
    * create a docstring at the top of the file, as in https://github.com/dib-lab/sourmash/blob/latest/src/sourmash/cli/compute.py
    * add the docstring and usage string to the subparser call, as in https://github.com/dib-lab/sourmash/blob/latest/src/sourmash/cli/compute.py
* verify that the usage string is output when you run the command with no arguments
* verify that the docstring is contained in the output of -h

Commands fixed:
- [x] sourmash gather
- [x] sourmash search
- [x] sourmash compare

NOTE:
The descriptions for all of the sourmash sig subcommands have been addressed in issue #1612 handled by PR #1714.

### CODE REVIEW

**Comment by keyabarve** [comment]
@ctb I added the docstring and usage for the `gather` command, but when I run `sourmash gather -h`, it does not produce the necessary information, instead it gives an error:
```
Traceback (most recent call last):
  File "/Users/keya/mambaforge/envs/sourmash_dev/bin/sourmash", line 33, in <module>
    sys.exit(load_entry_point('sourmash', 'console_scripts', 'sourmash')())
  File "/Users/keya/sourmash/src/sourmash/__main__.py", line 5, in main
    args = sourmash.cli.get_parser().parse_args(arglist)
  File "/Users/keya/sourmash/src/sourmash/cli/__init__.py", line 79, in parse_args
    args = super(SourmashParser, self).parse_args(args=args, namespace=namespace)
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 1818, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 1851, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 2042, in _parse_known_args
    positionals_end_index = consume_positionals(start_index)
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 2019, in consume_positionals
    take_action(action, args)
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 1928, in take_action
    action(self, namespace, argument_values, option_string)
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 1207, in __call__
    subnamespace, arg_strings = parser.parse_known_args(arg_strings, None)
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 1851, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 2060, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 2000, in consume_optional
    take_action(action, args, option_string)
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 1928, in take_action
    action(self, namespace, argument_values, option_string)
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 1092, in __call__
    parser.print_help()
  File "/Users/keya/sourmash/src/sourmash/cli/__init__.py", line 72, in print_help
    super(SourmashParser, self).print_help()
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 2548, in print_help
    self._print_message(self.format_help(), file)
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 2532, in format_help
    return formatter.format_help()
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 283, in format_help
    help = self._root_section.format_help()
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 214, in format_help
    item_help = join([func(*args) for func, args in self.items])
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 214, in <listcomp>
    item_help = join([func(*args) for func, args in self.items])
  File "/Users/keya/mambaforge/envs/sourmash_dev/lib/python3.9/argparse.py", line 300, in _format_usage
    usage = usage % dict(prog=self._prog)
ValueError: unsupported format character '`' (0x60) at index 1012
```

I think the error is being thrown at this line:
`1.4 Mbp      11.0%   58.0%      JANA01000001.1 Fusobacterium sp. OBRC...`
at the `%   58.0% ` part, but I'm not sure how to fix this. 

Could you provide any hints/suggestions?

**Comment by keyabarve** [comment]
@ctb I noticed that the problem was with the % sign, so I replaced all the % signs with the word "percent" since I couldn't find any other way to fix it. I'm not sure if that's the correct way though.

**Comment by keyabarve** [comment]
I also noticed that the sketch dna, protein, translate commands already have the doc and usage strings.

**Comment by keyabarve** [comment]
@ctb Could you please take a look at what I've written and let me know if I should change anything?

**Comment by ctb** [comment]
On Fri, Sep 03, 2021 at 06:19:36PM -0700, Keya Barve wrote:
> @ctb I noticed that the problem was with the % sign, so I replaced all the % signs with the word "percent" since I couldn't find any other way to fix it. I'm not sure if that's the correct way though.

try %% - and check out python string formatting :)

**Comment by keyabarve** [comment]
@ctb A few comments:
- I was able to fix the errors with the % signs. 
- I believe the descriptions for all the sourmash sig subcommands is being handled in #1714.
- I have added the descriptions for the `gather`, `search` and`compare` commands.
- I noticed that for sketch dna, translate and protein, the descriptions already exist in the respective files.

Please let me know if I need to add/change anything else.

**Comment by keyabarve** [comment]
@ctb Ready for review!

**Comment by keyabarve** [comment]
@ctb Please review.

**Comment by keyabarve** [comment]
@mr-eyes Please review.

**Comment by ctb** [comment]
> @mr-eyes Please review.

I think this one needs a lot of familiarity with sourmash use cases, so please wait to merge it until I review as well. But it could certainly use interim reviews from @mr-eyes and/or @hehouts :)

**Review by hehouts — COMMENTED** [review]
Looks great to me!

**Comment by ctb** [comment]
hi folks,

I was caught a little bit by surprise when I saw that the Markdown syntax (including links) was simply included in the help text, e.g.
```
[Jaccard index][3] or
(if signatures are created with `-p abund`) the [angular
similarity](https://en.wikipedia.org/wiki/Cosine_similarity#Angular_distance_and_similarity).
```

here it's worth noting a few things -
* `[3]` doesn't mean anything at all in this context - it's a numerical link reference to a link in the original doc file, doc/command-line.md.
* Markdown syntax isn't rendered at the command line, so in the help text output *on the command line* we should make sure it makes sense.

I'm working on making the necessary updates, but this is a plea to think about how the user is going to experience the changes and incorporate that into changes and/or reviews. thanks :)

**Comment by ctb** [comment]
It's also worth noting that while markdown->html rendering doesn't pay attention to line breaks, that is NOT true for the text output to command line. So reformatting things so that lines present properly is also worth doing.

**Comment by ctb** [comment]
Hi @keyabarve if you merge in https://github.com/sourmash-bio/sourmash/pull/1747, and everything looks good, we can merge this. Thank you!

**Comment by ctb** [comment]
hi @keyabarve also -

* please adjust the PR description at the top of this PR to be accurate and straightforward; this includes eliminating checkboxes that you don't plan on checking, changing the TODO to something like 'What we did', and referencing the right PR for sourmash sig commands, i.e. adding the issue and/or PR to '(Addressed in another issue)'. 
* reference this PR in a comment over on https://github.com/sourmash-bio/sourmash/issues/1557 stating that gather, search, and compare were handled in this PR, and that the sourmash sig commands are being handled elsewhere.

Basically any sourmash dev should be able to read the top of this PR to figure out what was done (not what you _intended_ to do, or _how it was done_ ;); and any sourmash dev should be able to look at #1557 and see what is left to be done and where it is being done. Thanks!

**Review by ctb — APPROVED** [review]
thank you! please merge if & when tests pass!

**Comment by keyabarve** [comment]
Will do! Thanks!

**Comment by ctb** [comment]
🎉
